src/openforms/emails/tests/test_send_mail_html.py

```diff
@@ -102,3 +106,75 @@ def test_oversize_content_raise_suspiciosu_operation(self):
                 "foo@sender.com",
                 ["foo@bar.baz"],
             )
+
+
+class DesignTokenFilterTest(TestCase):
+    """unit tests for ``emails.context._get_design_token_values``"""
+
+    def test_strip_px_unit(self):
+        self.design_tokens = {
+            "header-logo": {
+                "height": {"value": "100px"},
+                "width": {"value": "100px"},
+            }
+        }
+        result = _get_design_token_values(self.design_tokens)
+        self.assertEqual(result["logo"]["height"], "100px")
+        self.assertEqual(result["logo"]["width"], "100px")
+        self.assertEqual(result["logo"]["height_attr"], "100")
+        self.assertEqual(result["logo"]["width_attr"], "100")
+
+    def test_return_empty_string_for_non_px_unit(self):
+        self.design_tokens = {
+            "header-logo": {
+                "height": {"value": "100em"},
+                "width": {"value": "100em"},
+            }
+        }
+        result = _get_design_token_values(self.design_tokens)
+        self.assertEqual(result["logo"]["height"], "100em")
+        self.assertEqual(result["logo"]["width"], "100em")
+        self.assertEqual(result["logo"]["height_attr"], "")
+        self.assertEqual(result["logo"]["width_attr"], "")
+
+    def test_return_empty_string_for_auto_size(self):
+        """default for width: 'auto'"""
+        self.design_tokens = {"header-logo": {}}
+        result = _get_design_token_values(self.design_tokens)
+        self.assertEqual(result["logo"]["width"], "auto")
+        self.assertEqual(result["logo"]["width_attr"], "")
+
+
+class TemplateRenderingTest(TestCase):
+    """
+    integration test for the display of design token values
+
+    assumes that ``emails.context.get_wrapper_context`` works correctly,
+    hence the context variable is provided manually.
+    """
+
+    def test_design_token_values(self):
+        template = get_template("emails/wrapper.html")
+        ctx = {
+            "content": "<p>Hello there!</p>",
+            "main_website_url": "https://logoresizingdoneright.com",
+            "style": {
+                "logo": {
+                    "height": "150px",
+                    "width": "auto",
+                    "height_attr": "150",
+                    "width_attr": "",
+                }
+            },
+            "logo_url": "https://logo.png",
+        }
+        html_message = template.render(ctx)
+
+        img = PyQuery(html_message)("img")
+        img_outer_html = img.outer_html()
+
+        self.assertIn("height: 150px;", img_outer_html)
+        self.assertIn("width: auto;", img_outer_html)
+        self.assertIn('height="150"', img_outer_html)
+        self.assertNotIn('width="auto"', img_outer_html)
+        self.assertNotIn('width=""', img_outer_html)
```

Comment:
These assertions are still a bit too fuzzy - string comparisons/operations on HTML are usually a Bad Idea :tm: since you can have wrapping content that causes problems. What matters, is that the markup can be parsed correctly and results in the expected outcome, so we need to explicitly validate/test the attributes of the `img` element.

Using your PyQuery approach, we can do that this way:

```suggestion
        img = PyQuery(html_message)("img")

        self.assertEqual(img.attr("height"), "150")
        self.assertIsNone(img.attr("width"))

        img_styles = img.attr("style").replace(" ", "").split(";")
        self.assertIn("width:auto", img_styles)
        self.assertIn("height:150px", img_styles)
```
This checks three attributes explicitly: `width`, `height` and `style`. It abstracts away the potential single/double quote (mis)matches you can have, which the string assertions don't do.

The `style` attribute is then further checked - I noticed a lot of whitespace so cleaned up that string and then check for every expected style rule explicitly.